src/main/java/com/linkedin/replica/wall/commands/impl/EditCommentCommand.java

```diff
@@ -26,32 +26,32 @@ public Object execute() {
         WallHandler dbHandler = (WallHandler) this.dbHandler;
 
         // validate that all required arguments that are passed
-        validateArgs(new String[]{"commentId", "authorId", "parentPostId", "likesCount", "repliesCount", "images", "urls", "mentions", "text"});
+        validateArgs(new String[]{"commentId", "authorId", "parentPostId"});
 
 
         // call dbHandler to get error or success message from dbHandler
         Comment comment;
-        Gson googleJson = new Gson();
         String commentId = args.get("commentId").toString();
         String authorId = args.get("authorId").toString();
         String parentPostId = args.get("parentPostId").toString();
-        int likesCount = (int) args.get("likesCount");
-        int repliesCount = (int) args.get("repliesCount");
-        ArrayList<String> images = googleJson.fromJson((JsonArray) args.get("images"), ArrayList.class);
-        ArrayList<String> urls = googleJson.fromJson((JsonArray) args.get("urls"), ArrayList.class);
-        ArrayList<String> mentions = googleJson.fromJson((JsonArray) args.get("mentions"), ArrayList.class);
-        String text = args.get("text").toString();
 
         comment = dbHandler.getComment(commentId);

```

Comment:
Getting from the database again is not efficient. This is why we are changing the edits. One way to solve this is to set the commentId(using setCommentId). But I am not sure, so this needs to be tested.